openmc/mgxs/library.py

```diff
@@ -891,6 +897,9 @@ def get_xsdata(self, domain, xsdata_name, nuclide='total', xs_type='macro',
                       (type(None), bool))
         if tabular_points is not None:
             cv.check_greater_than('tabular_points', tabular_points, 1)
+        if subdomain is not None:
+            cv.check_iterable_type('subdomain', subdomain, Integral,
+                                   max_depth=2)
```

Comment:
Can you change this to `max_depth=3`. I have found the subdomains for a mesh require a max depth of 3.